plugins/modules/supported.py

```diff
@@ -0,0 +1,360 @@
+
+    # Mapping of parameter names to actual product names in the API
+    PRODUCT_NAME_MAPPING = {
+        "cloudera_manager_version": "Cloudera Manager",
+        "cloudera_runtime_version": "CDP Private Cloud Base",
+        "cloudera_data_services_version": "CDP Private Cloud Data Services",
+    }
+
+    def __init__(self, module):
+        """
+        Initialize the ClouderaSupportMatrix class.
+
+        Args:
+            module: AnsibleModule instance
+        """
+        self.module = module
+
+        # Extract product version parameters
+        self.product_versions = {}
+        for param_name in self.PRODUCT_NAME_MAPPING.keys():
+            version = module.params.get(param_name)
+            if version:
+                self.product_versions[param_name] = version
+
+        self.timeout = module.params.get("timeout", 30)
+
+        # Initialize return values
+        self.support_matrix_data = {}
+        self.filters_applied = {}
+        self.api_url = self.BASE_URL
+
+        # Execute the logic
+        self.process()
+
+    def _build_query_conditions(self):
+        """
+        Build query conditions for API filtering.
+
+        Returns:
+            str: Query conditions string for the API request
+        """
+        conditions = []
+
+        # Build single PRODUCT condition with comma-separated values
+        if self.product_versions:
+            products = []
+            for param_name, version in self.product_versions.items():
+                product_name = self.PRODUCT_NAME_MAPPING[param_name]
+                # URL encode spaces in both product name and version
+                # Use urllib.parse.quote for proper URL encoding
+                product_version_string = f"{product_name}-{version}"
+                encoded_product_version = quote(product_version_string)
+                products.append(encoded_product_version)
+                self.filters_applied[param_name] = version
+
+            if products:
+                # Join products with commas for a single PRODUCT parameter
+                product_condition = f"PRODUCT={','.join(products)}"
+                conditions.append(product_condition)
+
+        return ";".join(conditions)
+
+    def _build_api_url(self):
+        """
+        Build the complete API URL with query parameters.
+
+        Returns:
+            str: Complete API URL
+        """
+        conditions = self._build_query_conditions()
+
+        if conditions:
+            # Add trailing semicolon as shown in the curl example
+            # Don't URL encode the entire condition string, only spaces are encoded
+            self.api_url = f"{self.BASE_URL}?condition={conditions};"
+        else:
+            self.api_url = self.BASE_URL
+
+        return self.api_url
+
+    def process(self):
+        """
+        Fetch support matrix data from the Cloudera API using Ansible's fetch_url.
+
+        Returns:
+            bool: True if successful, False otherwise

```

Comment:
Remove

Reply:
Done in [1361398](https://github.com/cloudera-labs/cloudera.exe/pull/246/commits/13613986cd04ab928aa0e02f2e48aeae5b179ac2)